src/ucp/core/ucp_ep.c

```diff
@@ -365,6 +358,11 @@ void ucp_ep_release_id(ucp_ep_h ep)
 {
     ucs_status_t status;
 
+    if (ucp_ep_ext_control(ep)->local_ep_id == UCP_EP_ID_INVALID) {
+        ucs_assert(ep->flags & UCP_EP_FLAG_FAILED);
+        return;
+    }
+
```

Comment:
let's remove asserts in line 362 and 366
the function ucp_ep_release_id should not assume the context it's being used from